File: tests/machine/x/clj/break_continue.clj
(ns main)

(defn _equal [a b]
  (cond
    (and (sequential? a) (sequential? b))
      (and (= (count a) (count b)) (every? true? (map _equal a b)))
    (and (map? a) (map? b))
      (and (= (count a) (count b))
           (every? (fn [k] (_equal (get a k) (get b k))) (keys a)))
    (and (number? a) (number? b))
      (= (double a) (double b))
    :else
      (= a b)))

(declare numbers)

(defn -main []
  (def numbers [1 2 3 4 5 6 7 8 9]) ;; list of int
  (loop [_tmp0 (seq numbers)]
    (when _tmp0
      (let [n (clojure.core/first _tmp0)]
        (let [r (try
          (when (_equal (mod n 2) 0)
            (throw (ex-info "continue" {}))
          )
          (when (> n 7)
            (throw (ex-info "break" {}))
          )
          (println "odd number:" n)
          :next
        (catch clojure.lang.ExceptionInfo e
          (cond
            (= (.getMessage e) "continue") :next
            (= (.getMessage e) "break") :break
            :else (throw e))
          )
        )]
      (cond
        (= r :break) nil
        :else (recur (next _tmp0))
      )
    )
  )
)
)
)

(-main)

File: archived/tests/transpiler/x/pl/break_continue.pl
:- initialization(main).
:- style_check(-singleton).

main :-
    Numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9],
    N is 1,
    (false ->
),
    (false ->
),
    writeln("odd number: 1"),
    N1 is 2,
    (true ->
),
    (false ->
),
    writeln("odd number: 2"),
    N2 is 3,
    (false ->
),
    (false ->
),
    writeln("odd number: 3"),
    N3 is 4,
    (true ->
),
    (false ->
),
    writeln("odd number: 4"),
    N4 is 5,
    (false ->
),
    (false ->
),
    writeln("odd number: 5"),
    N5 is 6,
    (true ->
),
    (false ->
),
    writeln("odd number: 6"),
    N6 is 7,
    (false ->
),
    (false ->
),
    writeln("odd number: 7"),
    N7 is 8,
    (true ->
),
    (true ->
),
    writeln("odd number: 8"),
    N8 is 9,
    (false ->
),
    (true ->
),
    writeln("odd number: 9").
